### PR TITLE
B16 changes: IQ mixer and scan 1d

### PR DESCRIPTION
**Pull request by pieterjanstas**
I am going to say sorry in advance, this is a bit of a messy pull request, this branch has been running for a little too long in B16. These are the changes made in this branch:

a) IQ mixer optimization:
- fixed a bug where the wrong sine generator would be assigned to the wrong wave output if the I and Q output were on a different core (up until now the code had only been tested with IQ ports wave output 1&2 or 3&4).

b) scan_1d upgrades:
- added the ability to do a sawtooth scan (one way) in addition to the already existing triangle scan (back and forth). This can be set in the .json config file: if "sweep_type": "sawtooth", scan_1d will scan in one direction only. If "sweep_type": "triangle", scan_1d will scan in both directions. If nothing is specified, triangle sweep_type will be assumed.
- In the case that sweep_type": "sawtooth" is specified, the bottom window (usually used for the backwards scan) can be used for a second reading. If experiment() returns a scalar value, only the top window will be used to plot the return value. If experiment() returns a size-2 vector, the top window will be used to plot the first entry and the bottom window will be used to plot the second entry.
- IMPORTANT: THIS WILL AFFECT ALREADY WRITTEN EXPERIMENT CODE: Instead of having configure and experimeent take the arguments:

**def configure(clients):
    stuff

def experiment(frequency, clients, kwargs):
    return stuff**

It now takes the arguments:

**def configure(self):
    stuff

def experiment(frequency, self, kwargs):
    return stuff**

clients can still be accessed through self.clients. This way, all variables of the class can be accessed and changed in the configure and experiment functions.

As Can has already said, this is pushing the limits of the scan_1d code. Any fancier stuff should be done on the data_taker. However, it still is nice to have the simpler scan_1d be a bit more flexible.

### CODE REVIEW

**Comment by danielra97** [comment]
Did a quick look over, looked fine so far, will give a more careful look over later. In terms of merging I think since we might be warming up soonish in G12 might be best to wait till after (though wouldn't be a big deal either way should be a quick fix)

File: demo/T1_meas_notebook/T1.ipynb [hunk 1]
```diff
@@ -0,0 +1,774 @@
+{
```
Comment (cknaut) [inline demo/T1_meas_notebook/T1.ipynb:1]: I don't think measurement notebooks should be tracked by GitHub, I would remove it.

**Review by cknaut — APPROVED** [review]
Looks good, but I suggest removing the T1 notebook. @pieterjanstas , let me know if this works for you and I'll delete it from the git (make sure you have it backed up if it is relevant).

**Comment by pieterjanstas** [comment]
Yeah that makes sense, you can remove the T1 notebook.

**Review by cknaut — APPROVED** [review]
Looks good.

**Pull request merged**